storage/tianmu/core/temp_table.h

```diff
@@ -393,6 +396,7 @@ class TempTable : public JustATable {
   std::vector<JoinType> join_types;                 // vector of types of joins, one less than tables
   ParameterizedFilter filter;                       // multidimensional filter, contains multiindex,
                                                     // can be parametrized
+  bool filter_shallow = false;                      // is filter shallow
```

Comment:
makes the comment on `filter_shallow` more meaningful, just like ·shallow memcpy, reduce the comsuption of mem usage.·